ports/javascript/Makefile

```diff
@@ -16,7 +16,10 @@ INC += -I$(BUILD)
 CPP = clang -E
```

Comment:
To pass macros manually this needs to be:

```suggestion
CPP = clang -E -undef
```

Reply:
For some reason, GitHub refuses to let me comment on the remaining lines of the file.

```patch
-SRC_QSTR += $(SRC_C)
+SRC_QSTR += $(BUILD)/clang_predefs.h $(SRC_C)
+
+$(BUILD)/clang_predefs.h:
+       @emcc $(JSFLAGS) -E -x c /dev/null -dM > $@
```